pkg/kubectl/cmd/auth/cani.go

```diff
@@ -360,3 +369,20 @@ func printAccess(out io.Writer, rules []rbacv1.PolicyRule) error {
 	}
 	return nil
 }
+
+func isNamespaced(gvr schema.GroupVersionResource, discoveryClient discovery.DiscoveryInterface) (bool, error) {
+	apiResourceList, err := discoveryClient.ServerResourcesForGroupVersion(schema.GroupVersion{
+		Group: gvr.Group, Version: gvr.Version,
+	}.String())
+	if err != nil {
+		return true, err
+	}
+
+	for _, resource := range apiResourceList.APIResources {
+		if resource.Name == gvr.Resource {
+			return resource.Namespaced, nil
+		}
+	}
+
+	return true, fmt.Errorf("the server doesn't have a resource type '%s' in group '%s'", gvr.Resource, gvr.Group)
```

Comment:
in error cases, return zero-value for the boolean (`return false, fmt.Errorf(...`)